### ISSUE TITLE
Mount With Complex Password Hangs Forever

### ISSUE THREAD

**Issue by ansibot** [ansible/ansible#29642]
From @DaytonG on 2016-09-19T19:05:04Z

##### ISSUE TYPE

- Bug Report
##### COMPONENT NAME

mount
##### ANSIBLE VERSION

```
$ ansible --version
ansible 2.1.1.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```
##### CONFIGURATION

No changes were made to ansible configuration.
##### OS / ENVIRONMENT

ubuntu 12.04 (target host)
ubuntu 14.04 (ansible controller)
##### SUMMARY

When a mount password is provided that has unacceptable characters, fstab is not parsed properly which results in /bin/mount prompting for a password. This causes the mount command to hang silently forever.
##### STEPS TO REPRODUCE

We're mounting an NFS drive, which (eventually) uses AD as the auth mechanism. I don't believe that contributes to the issue. The whole issue seems to be that /bin/mount is asking for stdin.

```
- name: setup NFS host
  hosts: nfs_host
  vars: 
    mount_password: 'MyPassword/["HasFunnyCharacters'
  tasks:
    - name: mount nfs share
      mount:
        name: "/mnt/myshare"
        src: "//myshare/somedir"
        fstype: "cifs"
        opts: "domain=mydomain,user=myuser,password={{ mount_password }}"
        state: mounted
```

##### EXPECTED RESULTS

I believe that mount should timeout or die rather quickly if input is requested but unavailable.

I realize there are various other mount options to use to avoid this scenario, but it would be nice if the mount module handled the input request in some elegant way and fail out.
##### ACTUAL RESULTS

Mount will hang forever when `/bin/mount /mnt/myshare` is called.

Copied from original issue: ansible/ansible-modules-core#4915

**Comment by ansibot** [comment]
From @ansibot on 2016-09-19T19:05:04Z



@grossws ping, this issue is waiting for your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
From @ansibot on 2016-09-19T19:05:04Z

@ansible, ping. This issue is waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
From @grossws on 2016-09-19T19:05:04Z

EDIT: removed command -jctanner

**Comment by ansibot** [comment]
From @jctanner on 2016-09-19T19:05:04Z

@grossws what's broken?

**Comment by ansibot** [comment]
From @grossws on 2016-09-19T19:05:04Z

@jctanner, it seemed to be broken to me but it's because I thought it pinged me again. Just overlooked that last message in mail notifications thread pinged `@ansible` instead of me because both messages were near.

Sorry for false alarm.

**Comment by ansibot** [comment]
From @jctanner on 2016-09-19T19:05:04Z

@grossws no worries, thanks for wading through the mountain of emails I'm sure you get.

**Comment by biomassives** [comment]
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  acmeideal@gmail.com
    Domain biomassiv.es has exceeded the max emails per hour (100/100 (100%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
Received: from github-smtp2-ext6.iad.github.net ([192.30.252.197]:54674 helo=github-smtp2b-ext-cp1-prd.iad.github.net)
	by chi-server32.websitehostserver.net with esmtps (TLSv1.2:ECDHE-RSA-AES256-GCM-SHA384:256)
	(Exim 4.89)
	(envelope-from <noreply@github.com>)
	id 1drSqg-002QIV-R0
	for greg@biomassiv.es; Mon, 11 Sep 2017 12:50:18 -0500
Date: Mon, 11 Sep 2017 10:49:38 -0700
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=github.com;
	s=pf2014; t=1505152178;
	bh=/FZA1n9/ihF4WDXhvRWD2kF7kakr8jj5XZAMoqgvdzE=;
	h=From:Reply-To:To:Cc:In-Reply-To:References:Subject:List-ID:
	 List-Archive:List-Post:List-Unsubscribe:From;
	b=C8KuM/JUdvc2DCuNc8vbDliGm1AxSA46f76XLaABZZIG1JcTOVft9d/PY5e0IYPyD
	 b65PWA3GRSknL6dB9FqP5AncDGsbgyBXJAHToj+w+X1az4s5tbFwXz6H2dpWWC49HB
	 dO1udKc0/335zW821yXo4/+4mxgTUu5EchfJiw1k=

**Comment by ansibot** [comment]
cc @jtyr
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by jtyr** [comment]
I'm not sure if this should be handled by the module as the problematic input is hidden inside the `opts` parameter. What do you think, @abadger?

**Comment by infrascripting** [comment]
>I believe that mount should timeout or die rather quickly if input is requested but unavailable.
>
>I realize there are various other mount options to use to avoid this scenario, but it would be nice if the mount module handled the input request in some elegant way and fail out.

Related: https://github.com/ansible/ansible/issues/18305#issuecomment-327488669

**Comment by abadger** [comment]
If we can do something about it, I think we should.  Hanging is one of the
unfriendliest things we can do.

**Comment by jtyr** [comment]
I tried to search for allowed characters in the fstab opts but couldn't find any complete information about it. It looks like each filesystem can have its own set of allowed characters which makes it really hard to solve on the `mount` module level.

Other solution could be to introduce a `timeout` option for the `module.run_command()`. This [thread](https://stackoverflow.com/a/1191537) offers several solutions. @abadger what do you think?

**Comment by abadger** [comment]
A timeout sounds like a reasonable option if you can swing it.  Note that
I'm release manager for 2.4 so I don't have a lot of time to review the
strategies you might want to put into place.  But I can rubber stamp what
sounds like a good idea.

**Comment by ansibot** [comment]
cc @skvidal
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
Thank you very much for your interest in Ansible. Ansible has migrated much of the content into separate repositories to allow for more rapid, independent development. We are closing this issue/PR because this content has been moved to one or more collection repositories.

* lib/ansible/modules/system/mount.py -> https://galaxy.ansible.com/ansible/posix

For further information, please see:
https://github.com/ansible/ansibullbot/blob/master/docs/collection_migration.md
